src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
hmm, I am not sure if this is framed the best way. DataValueDeserializer indeed makes use of those constructors. But why would this library bother about some DataValueDeserializer? And, probably not in this case, in theory, there could be other users of those constructors than DataValueDeserializer.
Also, I note that builder callback for "wikibase-entityid" in Wikibase does use EntityIdValue::newFromArray :)

I know it is good to provide alternatives when deprecating a part of public interface, but maybe simple "DO NOT USE" would be enough here?